code/modules/paperwork/ticketmachine.dm

```diff
@@ -0,0 +1,201 @@
+//Bureaucracy machine!
+//Simply set this up in the hopline and you can serve people based on ticket numbers
+
+/obj/machinery/ticket_machine
+	name = "ticket machine"
+	icon = 'icons/obj/bureaucracy.dmi'
+	icon_state = "ticketmachine"
+	desc = "A marvel of bureaucratic engineering encased in an efficient plastic shell. It can be refilled with a hand labeler refill roll and linked to buttons with a multitool."
+	density = FALSE
+	maptext_height = 26
+	maptext_width = 32
+	maptext_x = 7
+	maptext_y = 10
+	layer = HIGH_OBJ_LAYER
+	var/ticket_number = 0 //Increment the ticket number whenever the HOP presses his button
+	var/current_number = 0 //What ticket number are we currently serving?
+	var/max_number = 100 //At this point, you need to refill it.
+	var/cooldown = 50
+	var/ready = TRUE
+	var/list/ticket_holders = list()
+	var/list/obj/item/ticket_machine_ticket/tickets = list()
+	var/id = 1
+
+/obj/machinery/ticket_machine/emag_act(mob/user) //Emag the ticket machine to dispense burning tickets, as well as randomize its number to destroy the HoP's mind.
+	if(emagged)
+		return
+	to_chat(user, "<span class='warning'>You overload [src]'s bureaucratic logic circuitry to its MAXIMUM setting.</span>")
+	ticket_number = rand(0,max_number)
+	current_number = ticket_number
+	emagged = TRUE
+	if(tickets.len)
+		for(var/obj/item/ticket_machine_ticket/ticket in tickets)
+			ticket.audible_message("<span class='notice'>\the [ticket] disperses!</span>")
+			qdel(ticket)
+		tickets.Cut()
+	update_icon()
+
+/obj/machinery/ticket_machine/Initialize(mapload)
+	. = ..()
+	update_icon()
+
+/obj/machinery/ticket_machine/proc/increment()
+	if(current_number > ticket_number)
+		return
+	if(current_number && !(emagged) && tickets[current_number])
+		tickets[current_number].audible_message("<span class='notice'>\the [tickets[current_number]] disperses!</span>")
+		qdel(tickets[current_number])
+	if(current_number < ticket_number)
+		current_number ++ //Increment the one we're serving.
+		playsound(src, 'sound/misc/announce_dig.ogg', 50, FALSE)
+		atom_say("Now serving ticket #[current_number]!")
+		if(!(emagged) && tickets[current_number])
+			tickets[current_number].audible_message("<span class='notice'>\the [tickets[current_number]] vibrates!</span>")
+		update_icon() //Update our icon here rather than when they take a ticket to show the current ticket number being served
+
+/obj/machinery/ticket_machine_button
+	name = "increment ticket counter"
+	desc = "Use this button after you've served someone to tell the next person to come forward."
+	icon = 'icons/obj/power.dmi'
+	icon_state = "crema_switch"
+	req_access = list()
+	var/id = 1
+	var/cooldown = FALSE
+
+/obj/machinery/ticket_machine_button/attack_ghost(mob/user)
+	if(user.can_advanced_admin_interact())
+		return attack_hand(user)
+
+/obj/machinery/ticket_machine_button/attack_hand(mob/user)
+	if(allowed(usr) || user.can_advanced_admin_interact())
+		for(var/obj/machinery/ticket_machine/M in world)
```

Comment:
`in world` is disgusting.

do a `GLOB.machines` search instead

Reply:
Fixed